tasks/src/main/scala/dagr/tasks/ScatterGather.scala

```diff
@@ -179,4 +209,46 @@ object ScatterGather {
       throw new UnsupportedOperationException("getTasks is not supported and should never be called on sub-scatters.")
     }
   }
+
+  /**
+    * A type of Scatter that is used to represent the first round of Scatter operation after the
+    * partitioner has created the partitions.  It's tasks are generated using the scatter partitions
+    * as input, and is also responsible for chaining together sub-scatters.
+    */
+  private class PrimaryGrouper[K, A, B <: Task](private val partitioner: Partitioner[A], f: A => K, g: (K, Seq[A]) => B) extends Pipeline with SubScatter[A,B] {
```

Comment:
This whole class can go away if you agree with the above comment about unsupporting a groupBy directly on a `PartitionerWrapper`.